core.go

```diff
@@ -122,3 +124,25 @@ type core struct {
 
 	fields map[string]interface{}
 }
+
+// follow same logic with sentry-go to filter unnecessary frames
+// ref:
+// https://github.com/getsentry/sentry-go/blob/362a80dcc41f9ad11c8df556104db3efa27a419e/stacktrace.go#L256-L280
+func filterFrames(frames []sentry.Frame) []sentry.Frame {
+	if len(frames) == 0 {
+		return nil
+	}
+	filteredFrames := make([]sentry.Frame, 0, len(frames))
+
+	for i := range frames {
+		// Skip zapsentry and zap internal frames, except for frames in _test packages (for
+		// testing).
+		if (strings.HasPrefix(frames[i].Module, "github.com/TheZeroSlave/zapsentry") ||
+			strings.HasPrefix(frames[i].Function, "go.uber.org/zap")) &&
+			!strings.HasSuffix(frames[i].Module, "_test") {
+			continue
```

Comment:
@kerkerj looks like we should use break here to remove all frames starting from N level

Reply:
@TheZeroSlave You mean we just drop the rest of those frames from `zapsentry`/`zap`, don't waste time to check again, am I right?

Reply:
@TheZeroSlave I've changed it to `break`.

Reply:
@kerkerj thnx!